connect-inject/endpoints_controller_test.go

```diff
@@ -1201,79 +1275,113 @@ func TestReconcileUpdateEndpoint(t *testing.T) {
 			expectedProxySvcInstances:  []*api.CatalogService{},
 		},
 	}
-	for _, tt := range cases {
-		t.Run(tt.name, func(t *testing.T) {
-			// The agent pod needs to have the address 127.0.0.1 so when the
-			// code gets the agent pods via the label component=client, and
-			// makes requests against the agent API, it will actually hit the
-			// test server we have on localhost.
-			fakeClientPod := createPod("fake-consul-client", "127.0.0.1", false)
-			fakeClientPod.Labels = map[string]string{"component": "client", "app": "consul", "release": "consul"}
+	for _, secure := range []bool{true, false} {
+		for _, tt := range cases {
+			t.Run(fmt.Sprintf("%s - secure: %v", tt.name, secure), func(t *testing.T) {
+				// The agent pod needs to have the address 127.0.0.1 so when the
+				// code gets the agent pods via the label component=client, and
+				// makes requests against the agent API, it will actually hit the
+				// test server we have on localhost.
+				fakeClientPod := createPod("fake-consul-client", "127.0.0.1", false)
+				fakeClientPod.Labels = map[string]string{"component": "client", "app": "consul", "release": "consul"}
 
-			// Create fake k8s client
-			k8sObjects := append(tt.k8sObjects(), fakeClientPod)
-			fakeClient := fake.NewClientBuilder().WithRuntimeObjects(k8sObjects...).Build()
+				// Create fake k8s client
+				k8sObjects := append(tt.k8sObjects(), fakeClientPod)
+				fakeClient := fake.NewClientBuilder().WithRuntimeObjects(k8sObjects...).Build()
 
-			// Create test consul server
-			consul, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
-				c.NodeName = nodeName
-			})
-			require.NoError(t, err)
-			defer consul.Stop()
-
-			consul.WaitForLeader(t)
-			consulClient, err := api.NewClient(&api.Config{
-				Address: consul.HTTPAddr,
-			})
-			require.NoError(t, err)
-			addr := strings.Split(consul.HTTPAddr, ":")
-			consulPort := addr[1]
+				masterToken := "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
+				caFile, certFile, keyFile := common.GenerateServerCerts(t)
+				// Create test consul server, with ACLs+TLS if necessary
+				consul, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+					if secure {
+						c.ACL.Enabled = true
+						c.ACL.DefaultPolicy = "deny"
+						c.ACL.Tokens.Master = masterToken
+						c.CAFile = caFile
+						c.CertFile = certFile
+						c.KeyFile = keyFile
+					}
+					c.NodeName = nodeName
+				})
+				require.NoError(t, err)
+				defer consul.Stop()
+				consul.WaitForSerfCheck(t)
```

Comment:
Is WaitForSerfCheck longer than WaitForLeader? If that makes more sense can we update Create/Delete tests to also use WaitForSerfCheck?

Reply:
Yeah WaitForSerfCheck waits the longest, ensures that we're able to replicate ACLs and read them back, its not totally necessary unless we're doing ACLs tho.

Reply:
Ahh I see, that makes sense why it's here but not for Create/Delete!